src/uu/du/src/du.rs

```diff
@@ -671,11 +673,21 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         read_files_from(file_from)?
     } else {
         match matches.get_one::<String>(options::FILE) {
-            Some(_) => matches
-                .get_many::<String>(options::FILE)
-                .unwrap()
-                .map(PathBuf::from)
-                .collect(),
+            Some(_) => {
+                let files = matches
+                    .get_many::<String>(options::FILE)
+                    .unwrap()
+                    .map(PathBuf::from);
+                if count_links {
+                    files.collect()
+                } else {
+                    // Deduplicate while preserving order
+                    let mut seen = std::collections::HashSet::new();
+                    files
+                        .filter(|path| seen.insert(path.clone()))
+                        .collect::<Vec<_>>()
+                }
+            }
```

Comment:
The `match` looks a bit strange. How about something like:
```rust
if let Some(files) = matches.get_many::<String>(options::FILE) {
    let files = files.map(PathBuf::from);
    if count_links {
        files.collect()
    } else {
        // Deduplicate while preserving order
        let mut seen = std::collections::HashSet::new();
        files
            .filter(|path| seen.insert(path.clone()))
            .collect::<Vec<_>>()
    }
} else {
    vec![PathBuf::from(".")]
}
```

Reply:
well done, much better :)